src/web3/contracts.ts

```diff
@@ -109,6 +116,30 @@ async function waitForContractKit(tries: number) {
   return contractKit
 }
 
+export function* getViemWallet(chain: Chain) {
+  if (viemWallets.has(chain)) {
+    return viemWallets.get(chain) as ViemWallet
+  }
+  const walletAddress = yield* select(walletAddressSelector)
+  const mnemonic = yield* call(getStoredMnemonic, walletAddress)
+  if (!mnemonic) {
+    throw new Error('No mnemonic found')
+  }
+  const keys = yield* call(generateKeysFromMnemonic, mnemonic)
```

Comment:
do we have to do this every time or could we use the getStoredPrivateKey method in KeychainLock?

Reply:
we could go the list stored accounts route... thats probably cleaner!

Reply:
You're right, it would be better to get from the keychain